### PR TITLE
Update menu docs

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Accounts for a few fixes including https://github.com/grammyjs/menu/issues/13, and adds more explanations.

### CODE REVIEW

**Review by habemuscode — APPROVED** [review]
LGTM

File: site/docs/plugins/menu.md [hunk 1]
```diff
@@ -444,6 +456,21 @@ Whenever a menu is sent, it will replay these operations to render your menu.
 This includes laying out all dynamic ranges and generating all dynamic labels.
 Once the menu is sent, the rendered button array will be forgotten again.
 
+When a menu is sent, every button contains callback query that stores the following information.
+
+- The menu identifier
+- The row/column position of the button
```
Comment (rojserbest) [inline site/docs/plugins/menu.md:462]:
```suggestion
- The row/column position of the button.
```

File: site/docs/plugins/menu.md [hunk 2]
```diff
@@ -444,6 +456,21 @@ Whenever a menu is sent, it will replay these operations to render your menu.
 This includes laying out all dynamic ranges and generating all dynamic labels.
 Once the menu is sent, the rendered button array will be forgotten again.
 
+When a menu is sent, every button contains callback query that stores the following information.
+
+- The menu identifier
```
Comment (rojserbest) [inline site/docs/plugins/menu.md:461]:
```suggestion
- The menu identifier.
```

File: site/docs/plugins/menu.md [hunk 3]
```diff
@@ -444,6 +456,21 @@ Whenever a menu is sent, it will replay these operations to render your menu.
...
+- The row/column position of the button
+- An optional payload
```
Comment (rojserbest) [inline site/docs/plugins/menu.md:463]:
```suggestion
- An optional payload.
```

File: site/docs/plugins/menu.md [hunk 4]
```diff
@@ -444,6 +456,21 @@ Whenever a menu is sent, it will replay these operations to render your menu.
...
+- A fingerprint flag that stores whether or not a fingerprint was used in the menu
```
Comment (rojserbest) [inline site/docs/plugins/menu.md:464]:
```suggestion
- A fingerprint flag that stores whether or not a fingerprint was used in the menu.
```

File: site/docs/plugins/menu.md [hunk 5]
```diff
@@ -444,6 +456,21 @@ Whenever a menu is sent, it will replay these operations to render your menu.
...
+- A 4-byte hash that encodes either the fingerprint, or the menu layout and the button label
```
Comment (rojserbest) [inline site/docs/plugins/menu.md:465]:
```suggestion
- A 4-byte hash that encodes either the fingerprint, or the menu layout and the button label.
```

**Review by rojserbest — CHANGES_REQUESTED** [review]

File: site/docs/plugins/menu.md [hunk 6]
```diff
@@ -408,28 +421,27 @@ const menu2 = new Menu("id", { onMenuOutdated: false });
...
-- the identifier of the menu,
-- the shape of the menu,
-- the position of the pressed button,
-- the payload, if specified, and
-- the text of the pressed button.
+We consider it outdated if:
 
-This data is effeciently compressed into a 4-byte hash that is stored in every button.
-It will then be compared to see if the menu is outdated before any handler is run.
+- the shape of the menu changed (number of rows, or number of buttons in any row)
```
Comment (WingLim) [inline site/docs/plugins/menu.md:426]: Upper the first word.
```suggestion
- The shape of the menu changed (number of rows, or number of buttons in any row)
```

**Review by WingLim — CHANGES_REQUESTED** [review]

File: site/docs/plugins/menu.md [hunk 7]
```diff
@@ -408,28 +421,27 @@ const menu2 = new Menu("id", { onMenuOutdated: false });
...
+- The row/column position of the pressed button is out of range.
+- The label the pressed button changed.
+- The pressed button does not contain a handler.
 
 It is possible that your menu changes, while all of the above things stay the same.
-This is usually not the case, but if you are creating a menu where this is likely, you should use a fingerprint function.
+It is also possible that your menu does not change fundamentally (i.e. the behavior of the handlers does not change), even though the above heuristic indicates indicates that the menu is outdates.
```
Comment (WingLim) [inline site/docs/plugins/menu.md:432]: Repeated words.
```suggestion
It is also possible that your menu does not change fundamentally (i.e. the behavior of the handlers does not change), even though the above heuristic indicates that the menu is outdates.
```

**Review by WingLim — CHANGES_REQUESTED** [review]

**Review by rojserbest — APPROVED** [review]

**Comment by rojserbest** [comment]
@WingLim you can approve too, if you have nothing left.

**Comment by KnorpelSenf** [comment]
@WingLim nice catches! Thanks for reviewing even though you said once you did not want to! :D

I was mainly requesting you for the translation.

**Review by WingLim — APPROVED** [review]
Good to go now.

**Pull request merged**